core/src/main/java/org/elasticsearch/action/admin/indices/warmer/delete/TransportDeleteWarmerAction.java

```diff
@@ -123,7 +124,7 @@ public ClusterState execute(ClusterState currentState) {
                     }
                 }
 
-                if (!globalFoundAtLeastOne) {
+                if (!globalFoundAtLeastOne && deleteAll == false) {
```

Comment:
you mind converting the fist conditional to use the `== false` too, just for consistency?